src/v/security/audit/audit_log_manager.cc

```diff
@@ -55,6 +55,13 @@ class audit_client {
 
     bool is_initialized() const { return _is_initialized; }
 
+    double percent_full() const {
+        auto max_avail
+          = config::shard_local_cfg().audit_client_max_buffer_size();
+        return 100.0
+               * (1.0 - (static_cast<double>(_send_sem.available_units()) / static_cast<double>(max_avail)));
```

Comment:
This would be better as a ratio.

Another problem is that `config::shard_local_cfg().audit_client_max_buffer_size()` can change between construction of the semaphore and calculating this value.

Reply:
>audit_client_max_buffer_size() can change between construction of the semaphore and calculating this value

Good point. Should cache its value on construction.